components/renku_data_services/project/db.py

```diff
@@ -142,15 +142,27 @@ async def insert_project(
         """Insert a new project entry."""
         if not session:
             raise errors.ProgrammingError(message="A database session is required")
-        ns = await self.group_repo.get_namespace_by_slug(user, project.namespace)
+        ns = await session.scalar(
+            select(schemas.NamespaceORM).where(schemas.NamespaceORM.slug == project.namespace.lower())
+        )
```

Comment:
So the reason I swapped the code here is that the `Namespace` object from the `group_repo` methods do not let me know the group or user id. I would have to do another db query to get it. I.e they get dropped when converting from ORM to model.

Reply:
Ok no problem. Just keep in mind this will not recognize or search "old" namespaces if the user has renamed their namespace. But I think that is acceptable, I think it makes sense to not allow people to create new projects by referencing an old/out-of-date namespace.